nirc_ehr/resources/queries/dbo/q_arrival.sql

```diff
@@ -1,29 +1,51 @@
 
-SELECT anm.ANIMAL_ID_NUMBER AS Id,
-       anm.BIRTH_DATE       AS birth,
-       anm.GENDER_ID        AS gender,
-       anm.SSB_ID           AS species,
-       NULL                 AS eventDate,
-       alt.NAME AS damSire,
-       alt.DESCRIPTION AS acqDateText,
-       'Alternate'          AS source
-FROM  ALTERNATE alt
-LEFT JOIN ANIMAL anm ON alt.ANIMAL_ID = anm.ANIMAL_ID
-WHERE alt.ALTERNATE_TYPE_ID = 7 -- 'Dam/Sire/Acq'
-  AND anm.ANIMAL_ID_NUMBER NOT LIKE 'A%' -- Animal born in centers are pre-appended with A's (in animal Id)
-  AND alt.DESCRIPTION IS NOT NULL
+        SELECT 'ALT' || alt.ALTERNATE_ID   AS objectid,
+               anm.ANIMAL_ID_NUMBER AS Id,
+               'unknown'      		AS performedby,
+               COALESCE(MAX(CAST(adt.CHANGE_DATETIME AS TIMESTAMP)), to_date('01/01/1970' ,'MM/DD/YYYY')) AS modified,
+               anm.BIRTH_DATE       AS birth,
+               anm.GENDER_ID        AS gender,
+               anm.SSB_ID.SPECIES_ID           AS species,
+               NULL                 AS eventDate,
+               alt.NAME             AS damSire,
+               alt.DESCRIPTION      AS acqDateText,
+               'Alternate'          AS source
+        FROM  ALTERNATE alt
+        LEFT JOIN ANIMAL anm ON alt.ANIMAL_ID = anm.ANIMAL_ID
+        LEFT JOIN AUDIT_TRAIL adt ON alt.ALTERNATE_ID = CAST(substring(PRIMARY_KEY_VALUES, length('Alternate_ID = ')) AS INTEGER)
+            AND adt.TABLE_NAME = 'ALTERNATE'
+            AND adt.COLUMN_NAME != 'DELETE'
```

Comment:
do we get any ALTERNATE IDs that are present in AUDIT_TRAIL and in ALTERNATE table where the COLUMN_NAME = 'DELETE'? My understanding is that if there is a row present in AUDIT_TRAIL and has COLUMN_NAME = 'DELETE', it won't be present in corresponding TABLE_NAME which would make this join unnecessary.

Reply:
Good call. Not  needed, exact same results without it.